exchanges/poloniex/poloniex_websocket.go

```diff
@@ -663,3 +627,515 @@ func (p *Poloniex) wsSendAuthorisedCommand(command string) error {
+		return "lending"
+	default:
+		return "unknown"
+	}
+}
+
+func (p *Poloniex) processAccountTrades(notification []interface{}) error {
+	if len(notification) < 11 {
+		return errNotEnoughData
+	}
+
+	tradeID, ok := notification[1].(float64)
+	if !ok {
+		return fmt.Errorf("%w tradeID not float64", errTypeAssertionFailure)
+	}
+
+	r, ok := notification[2].(string)
+	if !ok {
+		return fmt.Errorf("%w rate not string", errTypeAssertionFailure)
+	}
+	rate, err := strconv.ParseFloat(r, 64)
+	if err != nil {
+		return err
+	}
+
+	a, ok := notification[3].(string)
+	if !ok {
+		return fmt.Errorf("%w amount not string", errTypeAssertionFailure)
+	}
+	amount, err := strconv.ParseFloat(a, 64)
+	if err != nil {
+		return err
+	}
+
+	// notification[4].(string) is the fee multiplier
+	// notification[5].(string) is the funding type 0 (exchange wallet),
+	// 1 (borrowed funds), 2 (margin funds), or 3 (lending funds)
+
+	orderID, ok := notification[6].(float64)
+	if !ok {
+		return fmt.Errorf("%w orderID not float64", errTypeAssertionFailure)
+	}
+
+	fee, ok := notification[7].(string)
+	if !ok {
+		return fmt.Errorf("%w fee not string", errTypeAssertionFailure)
+	}
+	totalFee, err := strconv.ParseFloat(fee, 64)
+	if err != nil {
+		return err
+	}
+
+	t, ok := notification[8].(string)
+	if !ok {
+		return fmt.Errorf("%w time not string", errTypeAssertionFailure)
+	}
+	timeParse, err := time.Parse(common.SimpleTimeFormat, t)
+	if err != nil {
+		return err
+	}
+
+	// null returned so ok check is not needed
+	clientOrderID, _ := notification[9].(string)
+
+	tt, ok := notification[10].(string)
+	if !ok {
+		return fmt.Errorf("%w time not string", errTypeAssertionFailure)
+	}
+	tradeTotal, err := strconv.ParseFloat(tt, 64)
+	if err != nil {
+		return err
+	}
+
+	p.Websocket.DataHandler <- &order.Modify{
+		Exchange: p.Name,
+		ID:       strconv.FormatFloat(orderID, 'f', -1, 64),
+		Fee:      totalFee,
+		Trades: []order.TradeHistory{{
+			Price:     rate,
+			Amount:    amount,
+			Fee:       totalFee,
+			Exchange:  p.Name,
+			TID:       strconv.FormatFloat(tradeID, 'f', -1, 64),
+			Timestamp: timeParse,
+			Total:     tradeTotal,
+		}},
+		AssetType:     asset.Spot,
+		ClientOrderID: clientOrderID,
+	}
+	return nil
+}
+
+func (p *Poloniex) processAccountKilledOrder(notification []interface{}) error {
+	if len(notification) < 3 {
+		return errNotEnoughData
+	}
+
+	orderID, ok := notification[1].(float64)
+	if !ok {
+		return fmt.Errorf("%w order ID not float64", errTypeAssertionFailure)
+	}
+
+	// null returned so ok check is not needed
+	clientOrderID, _ := notification[2].(string)
+
+	p.Websocket.DataHandler <- &order.Modify{
+		Exchange:      p.Name,
+		ID:            strconv.FormatFloat(orderID, 'f', -1, 64),
+		Status:        order.Cancelled,
+		AssetType:     asset.Spot,
+		ClientOrderID: clientOrderID,
+	}
+	return nil
+}
+
+func (p *Poloniex) processTrades(currencyID float64, subData []interface{}) error {
+	if !p.IsSaveTradeDataEnabled() {
```

Comment:
This is preventing test coverage. Could you please have `p.SetSaveTradeDataStatus(true)` for the relevant test function?